spec.html

```diff
@@ -6726,27 +6869,69 @@ <h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-definefield" type="abstract operation">
+    <emu-clause id="sec-initializeprivatemethods" type="abstract operation">
       <h1>
-        DefineField (
+        InitializePrivateMethods (
+          _O_: an Object,
+          _elementDefinitions_: a List of ClassElementDefinition Records,
+        ): either a normal completion containing ~unused~ or a throw completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _privateMethods_ be a new empty List.
+        1. For each element _element_ of _elementDefinitions_, do
+          1. If _element_.[[Key]] is a Private Name and _element_.[[Kind]] is one of ~method~, ~getter~, ~setter~, or ~accessor~, then
+            1. If _element_.[[Kind]] is ~method~, then
+              1. Let _privateElement_ be PrivateElement { [[Key]]: _element_.[[Key]], [[Kind]]: ~method~, [[Value]]: _element_.[[Value]] }.
+              1. Append _privateElement_ to _privateMethods_.
+            1. Else if _element_.[[Kind]] is ~accessor~, then
+              1. Let _privateElement_ be PrivateElement { [[Key]]: _element_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _element_.[[Get]], [[Set]]: _element_.[[Set]] }.
+              1. Append _privateElement_ to _privateMethods_.
+            1. Else,
+              1. Assert: _element_.[[Kind]] is either ~getter~ or ~setter~.
+              1. If _element_.[[Kind]] is ~getter~, let _getter_ be _element_.[[Get]].
+              1. Else, let _getter_ be *undefined*.
+              1. If _element_.[[Kind]] is ~setter~, let _setter_ be _element_.[[Set]].
+              1. Else, let _setter_ be *undefined*.
+              1. Let _existing_ be *undefined*.
+              1. If _privateMethods_ contains a PrivateElement _e_ such that _e_.[[Key]] is _element_.[[Key]], then
+                1. Assert: _e_.[[Kind]] is ~accessor~.
+                1. Set _existing_ to _e_.
+                1. If _e_.[[Get]] is not *undefined*, set _getter_ to _existing_.[[Get]].
+                1. If _e_.[[Set]] is not *undefined*, set _setter_ to _existing_.[[Set]].
+              1. Let _privateElement_ be PrivateElement { [[Key]]: _element_.[[Key]], [[Kind]]: ~accessor~, [[Get]]: _getter_, [[Set]]: _setter_ }.
+              1. If _existing_ is not *undefined*, replace _existing_ in _privateMethods_ with _privateElement_.
```

Comment:
I understand that steps 6 through 9 define a no-op if there is already a private accessor pair with the same key, is this correct? If so, could we reword 7.c and 7.d? It seems a little strange to use `_e_` and `_existing_` in the same line when `_existing_` was set to `_e_` in the previous step.